services/wallet/api.go

```diff
@@ -589,7 +591,67 @@ func (api *API) AddressExists(ctx context.Context, address types.Address) (bool,
 	return api.s.accountsDB.AddressExists(address)
 }
 
-// Returns details for the passed address (response doesn't include derivation path)
+// AddressDetails returns details for passed params (passed address, chains to check, timeout for the call to complete)
+// if chainIDs is empty, it will use all active chains
+// if timeout is zero, it will wait until the call completes
+// response doesn't include derivation path
+func (api *API) AddressDetails(ctx context.Context, params *requests.AddressDetails) (*DerivedAddress, error) {
```

Comment:
Thanks for the change 👍 

Just one thing: the argument should be a json string, to be able to pass it from the client. Like here:
https://github.com/status-im/status-go/blob/2a7abcc760e42f89a4d8d44998b52c3902c26ba5/mobile/status.go#L325-L330

Reply:
@igor-sirotin no need for this, cause this function belongs to the wallet service, and handling calls via `callPrivateRPC` function unmarshal data automatically to the struct type.